neonvm/controllers/virtualmachine_qmp_queries.go

```diff
@@ -368,8 +451,11 @@ func QmpUnplugMemory(ip string, port int32) error {
 	var merr error
 	for i = plugged - 1; i >= 0; i-- {
 		// remove pc-dimm device
-		cmd := []byte(fmt.Sprintf(`{"execute": "device_del", "arguments": {"id": "%s"}}`, memoryDevices[i].Data.Id))
-		_, err = mon.Run(cmd)
+		idx, err := QmpMatchMemBackend(memoryDevices[i].Data.Memdev)
+		if err != nil {
+			merr = errors.Join(merr, err)
+		}
+		err = QmpDelMemoryDevice(mon, idx)
```

Comment:
not a blocker:
The `continue` just below this seems... scary to me.

IIRC I've seen unplug "timeout" (from QMP's perspective) but eventually occur inside the VM if the guest's ACPI handler just takes a while to migrate pages.
If that happens, is it possible to accidentally unplug many memory slots? i.e. start unplugging many memory slots because none of them have succeeded yet, and then end up unplugging more than you intended?

Reply:
Good point. 

I'd rather keep it as-is in this commit. I've addressed it in the next one, where this logic gets overhauled.


It is modified  such that we attempt to unplug no more than the intended number of slots.